Dịch preliminaries/linear-algebra phần 9 (#368)
aivivn/d2l-vn: chapter_preliminaries/linear-algebra_vn.md

## Matrix-Matrix Multiplication
-->

## Phép nhân ma trận

<!--
If you have gotten the hang of dot products and matrix-vector products, then *matrix-matrix multiplication* should be straightforward.
-->

> Nếu bạn đã quen thuộc với tích của ma trận và vector, tích của hai ma trận cũng tương tự như thế.

Reviewer: ```suggestion
Nếu bạn đã quen thuộc với tích vô hướng và tích ma trận-vector, tích của hai ma trận cũng tương tự như thế.
```

<!--
Say that we have two matrices $\mathbf{A} \in \mathbb{R}^{n \times k}$ and $\mathbf{B} \in \mathbb{R}^{k \times m}$:
-->

> Giải sử ta có hai ma trận $\mathbf{A} \in \mathbb{R}^{n \times k}$ và $\mathbf{B} \in \mathbb{R}^{k \times m}$:

Reviewer: ```suggestion
Giả sử ta có hai ma trận $\mathbf{A} \in \mathbb{R}^{n \times k}$ và $\mathbf{B} \in \mathbb{R}^{k \times m}$:
```

\vdots & \vdots & \ddots & \vdots \\
 b_{k1} & b_{k2} & \cdots & b_{km} \\
\end{bmatrix}.$$

<!--
Denote by $\mathbf{a}^\top_{i} \in \mathbb{R}^k$ the row vector representing the $i^\mathrm{th}$ row of the matrix $\mathbf{A}$, and let $\mathbf{b}_{j} \in \mathbb{R}^k$ be the column vector from the $j^\mathrm{th}$ column of the matrix $\mathbf{B}$.
To produce the matrix product $\mathbf{C} = \mathbf{A}\mathbf{B}$, it is easiest to think of $\mathbf{A}$ in terms of its row vectors and $\mathbf{B}$ in terms of its column vectors:
-->

> Đặt $\mathbf{a}^\top_{i} \in \mathbb{R}^k$ là vector hàng biểu diễn hàng thứ $$ của ma trận $\mathbf{A}$ và $\mathbf{b}_{j} \in \mathbb{R}^k$ là vector cột thứ $j$ của ma trận $\mathbf{B}$.

Reviewer: ```suggestion
Đặt $\mathbf{a}^\top_{i} \in \mathbb{R}^k$ là vector hàng biểu diễn hàng thứ $i$ của ma trận $\mathbf{A}$ và $\mathbf{b}_{j} \in \mathbb{R}^k$ là vector cột lấy từ cột thứ $j$ của ma trận $\mathbf{B}$.
```

Author: Mình nghĩ không cần, vì khái niệm "vector cột", "vector hàng" đã được dùng phổ biến.

> Để tính ma trận tích $\mathbf{C} = \mathbf{A}\mathbf{B}$, đơn giản nhất là viết các hàng của ma trận $\mathbf{A}$ các cột của ma trận $\mathbf{B}$:

Reviewer: ```suggestion
Để tính ma trận tích $\mathbf{C} = \mathbf{A}\mathbf{B}$, cách đơn giản nhất là viết các hàng của ma trận $\mathbf{A}$ các cột của ma trận $\mathbf{B}$:
```

Then the matrix product $\mathbf{C} \in \mathbb{R}^{n \times m}$ is produced as we simply compute each element $c_{ij}$ as the dot product $\mathbf{a}^\top_i \mathbf{b}_j$:
-->

> Khi đó ma trận tích $\mathbf{C} \in \mathbb{R}^{n \times m}$ được tạo bằng cách tính mỗi phần tử $c_{ij}$ như tích vô hướng  $\mathbf{a}^\top_i \mathbf{b}_j$:

Reviewer: ```suggestion
Khi đó ma trận tích $\mathbf{C} \in \mathbb{R}^{n \times m}$ được tạo bằng cách đặt mỗi phần tử $c_{ij}$ là tích vô hướng $\mathbf{a}^\top_i \mathbf{b}_j$:
```

> Khi đó ma trận tích $\mathbf{C} \in \mathbb{R}^{n \times m}$ được tạo bằng cách tính mỗi phần tử $c_{ij}$ như tích vô hướng  $\mathbf{a}^\top_i \mathbf{b}_j$:

Author: ```suggestion
Khi đó ma trận tích $\mathbf{C} \in \mathbb{R}^{n \times m}$ được tạo bằng cách tính mỗi phần tử $c_{ij}$ bằng tích vô hướng $\mathbf{a}^\top_i \mathbf{b}_j$:
```

> Khi đó ma trận tích $\mathbf{C} \in \mathbb{R}^{n \times m}$ được tạo bằng cách tính mỗi phần tử $c_{ij}$ bằng tích vô hướng $\mathbf{a}^\top_i \mathbf{b}_j$:

Author: ```suggestion
Khi đó ma trận tích $\mathbf{C} \in \mathbb{R}^{n \times m}$ được tạo bằng cách tính từng phần tử $c_{ij}$ của nó bằng tích vô hướng $\mathbf{a}^\top_i \mathbf{b}_j$:
```

After multiplication, we obtain a matrix with $5$ rows and $3$ columns.
-->

> Ta có thể coi tích hai ma trận $\mathbf{AB}$ như việc tính $m$ phép nhân ma trận và vector sau đó ghép các kết quả với nhau để tạo ra một ma trận $n \times m$.

Reviewer: ```suggestion
Ta có thể coi tích hai ma trận $\mathbf{AB}$ như việc tính $m$ phép nhân ma trận và vector, sau đó ghép các kết quả với nhau để tạo ra một ma trận $n \times m$.
```

> Giống như tích vô hướng và phép nhân ma trận và vector, ta có thể tính phép nhân hai ma trận bằng cách sử dụng hàm `dot`.

Reviewer: ```suggestion
Giống như tích vô hướng và phép nhân ma trận-vector, ta có thể tính phép nhân hai ma trận bằng cách sử dụng hàm `dot`.
```

*

<!-- Phần 9 -->

> * Vũ Hữu Tiệp

Author: ```suggestion
* Vũ Hữu Tiệp
* Phạm Minh Đức
```
